lib/Runtime/Base/WindowsGlobalizationAdapter.cpp

```diff
@@ -829,20 +829,41 @@ if (this->object) \
         return ResolveLocaleLookup(scriptContext, locale, resolved);
     }
 
-    int IcuIntlAdapter::GetUserDefaultLocaleName(_Out_ LPWSTR lpLocaleName, _In_ int cchLocaleName)
+    size_t IcuIntlAdapter::GetUserDefaultLanguageTag(_Out_ char16* langtag, _In_ size_t cchLangtag)
     {
-        // XPLAT-TODO (doilij): implement GetUserDefaultLocaleName
-        const auto locale = _u("en-US");
-        const size_t len = 5;
-        if (lpLocaleName)
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+        char bcp47[ULOC_FULLNAME_CAPACITY];
+        char defaultLocaleId[ULOC_FULLNAME_CAPACITY];
+
+        int32_t written = uloc_getName(nullptr, defaultLocaleId, ULOC_FULLNAME_CAPACITY, &error);
+        if (U_FAILURE(error) || written <= 0 || written >= ULOC_FULLNAME_CAPACITY)
```

Comment:
> if (U_FAILURE(error) || written <= 0 || written >= ULOC_FULLNAME_CAPACITY) [](start = 7, length = 75)

nit: success branch first for readability, unless it's easier to express the failure case in an `if` conditional.
To work around this, personally I will sometimes declare a boolean named (something like) `success` or `failed` and then negate if necessary depending on what I actually want to express for the if condition.